lib/utils/transformpipeline/transforms.py

```diff
@@ -287,6 +288,8 @@ def __init__(self):
     def transform_value(self, entry: dict) -> dict:
         entry['sequence'] = entry['sequence'].replace('\n', '')
         entry['length'] = len(entry['sequence'])
+        lineage_dict = json.loads(entry['pango_lineage'])
+        entry['pango_lineage'] = lineage_dict[0]['lineage']
```

Comment:
Out of interest, what does `entry['pango_lineage']` look like?

Reply:
Sure thing! From [here](https://github.com/robert-koch-institut/SARS-CoV-2-Sequenzdaten_aus_Deutschland?tab=readme-ov-file#variablen-und-werte), the original format looks like

```
[{'method': 'PANGOLIN_LATEST', 'classification_version': 'PUSHER-v1.28.1', 'tool_version': '4.3', 'lineage': 'BA.2', '@qc_notes': 'Ambiguous_content:0.02', '@is_designated': False, '@qc_status': 'pass', '@conflict': 0.0, '@note': 'Usher placements: BA.2(1/1)'}]
```

Or a few examples from the data:

|pango_lineages|
|---|
|[{"classification_version": "PUSHER-v1.29", "lineage": "BA.2", "tool_version": "4.3", "method": "PANGOLIN_LATEST", "@is_designated": false, "@qc_status": "pass", "@conflict": 0.0, "@qc_notes": "Ambiguous_content:0.02", "@note": "Usher placements: BA.2(1/1)"}]|
|[{"classification_version": "PUSHER-v1.29", "lineage": "BA.2.9", "tool_version": "4.3", "method": "PANGOLIN_LATEST", "@is_designated": false, "@qc_status": "pass", "@conflict": 0.0, "@qc_notes": "Ambiguous_content:0.02", "@note": "Usher placements: BA.2.9(1/1)"}]|
|[{"classification_version": "PUSHER-v1.29", "lineage": "BA.2", "tool_version": "4.3", "method": "PANGOLIN_LATEST", "@is_designated": false, "@qc_status": "pass", "@conflict": 0.0, "@qc_notes": "Ambiguous_content:0.02", "@note": "Usher placements: BA.2(2/2)"}]

which, after the change, get corrected to:

|pango_lineages|
|---|
|"BA.2"|
|"BA.2.9"|
|"BA.2"|

Reply:
Thanks! My only suggestion would be to do something like
```py
assert len(lineage_dict)==1, f"RKI pango_lineage unexpectedly had more than one entry. Strain: {entry['name']}"
# (not sure if "name" is the correct key to use)
```
(also, perhaps `lineage_dict` isn't the best name if it's a list)

Reply:
I might rename `lineage_dict` to "lineage_json_blob" based on their own description (`JSON-Blob`).

Instead of erroring out on multiple entries, I would move to looping through the entries to find the one with `"method": "PANGOLIN_LATEST"`. We're not sure if they're setting this up to historical past pangolin lineage assignments, as lineage systems get updated.

Reply:
Heh, in the end, I went with your assert statement to at least flag when multiple entries start showing up (https://github.com/nextstrain/ncov-ingest/pull/476/commits/e57f3c0ff53ba134e54b7dca0d9b759e1c653658). This will help us double-check the purpose of the multiple entries, if they do eventually occur.

I double-checked `entry.keys()`, and the `entry['rki_accession']` is the strain ID.

Reply:
Hi, I just stumbled upon this issue, and we intend always to use PANGOLIN_LATEST for the latest assignment. So, you can filter the JSON array this way. I am sorry for not being more specific in our README. I will change that in our documentation. Please feel free to write us an email if you have questions igs-developers(at)rki.de .

Reply:
Thanks for chiming in @TheBready! 

Tracking in https://github.com/nextstrain/ncov-ingest/issues/478